### PR TITLE
Travis Runs Unit Tests and Integration Tests

### PR DESCRIPTION
**Pull request by mtomcal**
- Built a working travis config with working integration and unit test stages run in parallel
- Found a bad unit test that failed in Travis and corrected it. It was misusing exports without understanding `require`-ing an `export default` versus `export`.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
   - "12"
 services:
-  - docker
\ No newline at end of file
+  - docker
+before_install:
+  # Create a master branch for commitlint
+  # https://github.com/conventional-changelog/commitlint/issues/6
+  - git remote set-branches --add origin master && git fetch
+install: NODE_ENV=development npm ci
```
Comment (mtomcal) [inline .travis.yml]: I need `rimraf` to be installed for the `postinstall` activities

File: __tests__/server/utils/logging/development-formatters/index.spec.js [hunk 2]
```diff
@@ -14,50 +14,41 @@
  * permissions and limitations under the License.
  */
 
-import yargs from 'yargs';
-
-jest.mock('yargs', () => ({ argv: {} }));
+jest.mock('../../../../../src/server/utils/logging/development-formatters/verbose', () => 'verbose export');
+jest.mock('../../../../../src/server/utils/logging/development-formatters/friendly', () => 'friendly export');
```
Comment (mtomcal) [inline __tests__/server/utils/logging/development-formatters/index.spec.js]: `./src/server/utils/logging/development-formatters/index.js` conditionally exports `verbose` or `friendly` and I am just testing that branching logic. Mocking away the actual contents of the export as they are not needed for this test case.

File: __tests__/server/utils/logging/development-formatters/index.spec.js [hunk 3]
```diff
@@ -14,50 +14,41 @@
...
-      afterWrite: jest.fn(),
-    }));
-    jest.mock('../../../../../src/server/utils/logging/development-formatters/friendly', () => ({
-      formatter: jest.fn(),
-      beforeWrite: jest.fn(),
-      afterWrite: jest.fn(),
-    }));
+  });
 
-    verbose = require('../../../../../src/server/utils/logging/development-formatters/verbose').default;
```
Comment (mtomcal) [inline __tests__/server/utils/logging/development-formatters/index.spec.js]: `default` is undefined in `verbose.js` and `friendly.js` and it only does `export` not `export default`!

File: package.json [hunk 4]
```diff
@@ -24,7 +24,7 @@
     "start:prod-sample": "docker-compose -f ./prod-sample/docker-compose.yml up --abort-on-container-exit",
     "test:lint": "eslint --ext js,jsx,md,snap .",
     "start": "node lib/server/index.js",
-    "test:unit": "jest --testPathIgnorePatterns integration --config jest.config.js",
+    "test:unit": "cross-env NODE_ENV=development jest --testPathIgnorePatterns integration --config jest.config.js",
```
Comment (mtomcal) [inline package.json]: Not sure if this is what we want to do long term.

File: .travis.yml [hunk 5]
```diff
@@ -1,5 +1,19 @@
...
+jobs:
+  include:
+    - stage: "Tests"
+      name: "Unit Tests"
+      script: npm run test:lint && npm run test:unit && npm run test:danger && npm run test:git-history
+    - script: npm run test:integration
+      name: "Integration Tests"
```
Comment (10xLaCroixDrinker) [inline .travis.yml]: can you put name first here?

**Pull request merged**